common/models/templates/test/trie-compression.js

```diff
@@ -0,0 +1,324 @@
+      });
+
+      it('decompresses two-char strings of one-char value width', () => {
+        assert.equal(decompressNumber('\u0000 ', 0), 0x0020);
+      });
+    });
+
+    describe('with mock-inlining', () => {
+      it('decompresses single-char strings', () => {
+        assert.equal(decompressNumber('xxx xx', 3, 4), 0x0020);
+        assert.equal(decompressNumber('xx"x', 2, 3), '"'.charCodeAt(0));
+        assert.equal(decompressNumber('\uffff\ufffe', 1), 0xfffe);
+      });
+
+      it('decompresses two-char strings', () => {
+        assert.equal(decompressNumber('xxxx  xx', 4, 6), 0x00200020);
+      });
+    });
+  });
+
+  describe('`Entry`s', () => {
+    it('not inlined', () => {
+      const mockedDecompression = TEST_DATA.ENTRIES.four.decompressed;
+      const compressionSrc = TEST_DATA.ENTRIES.four.compressed;
+      assert.deepEqual(decompressEntry(compressionSrc), mockedDecompression);
+    });
+
+    it('inlined', () => {
+      const mockedDecompression = TEST_DATA.ENTRIES.four.decompressed;
+
+      // total length: header = 5, text = 8 -> 13.
+      const compressionSrc = `xxxxx${TEST_DATA.ENTRIES.four.compressed}xx`;
+      assert.deepEqual(decompressEntry(compressionSrc, /* start index */ 5), mockedDecompression);
+    });
+  });
+
+  describe('Leaf nodes', () => {
+    describe('bootstrapping cases', () => {
+      it('not inlined', () => {
+        const encodedLeaf = TEST_DATA.LEAVES.four.compressed;
+        assert.deepEqual(decompressNode(encodedLeaf, 0), TEST_DATA.LEAVES.four.decompressed);
+      });
+
+      it('inlined', () => {
+        const encodedLeaf = TEST_DATA.LEAVES.four.compressed;
+        assert.deepEqual(decompressNode(`xxxxxxxxx${encodedLeaf}xx`, 9), TEST_DATA.LEAVES.four.decompressed);
+      });
+    });
+  });
+
+  describe('Internal nodes', () => {
+    describe('bootstrapping cases', () => {
+      it('not inlined', () => {
+        const encodedNode = TEST_DATA.NODES.four.compressed;
+        assert.deepEqual(decompressNode(encodedNode, 0), TEST_DATA.NODES.four.decompressed);
+      });
+
+      it('inlined', () => {
+        const encodedNode = TEST_DATA.NODES.four.compressed;
+        assert.deepEqual(decompressNode(`xxxxxxx${encodedNode}xx`, 7), TEST_DATA.NODES.four.decompressed);
+      });
+    });
+  });
+
+  it('compresses fixture successfully: english-1000', () => {
+    const trie = jsonFixture('tries/english-1000');
+    // The test:  does it throw?
+    const compressedTrie = compressNode(trie.root);
+
+    const encodedSerializableString = `${
+      compressedTrie
+        .replace(/\\/g, '\\\\') // preservation - escape char as literal
+        .replace(/`/g, '\\`') // escape the primary quote
+        .replace(/\n/g, '\\n') // prevent CRLF shenanigans
+        .replace(/\r/g, '\\r')
+    }`;
+
+    const compression = {
+      // Achieves FAR better compression than JSON.stringify, which \u-escapes most chars.
```

Comment:
Unless using `ENCODED_NUM_BASE=0x0020` or similar.  `JSON.stringify` likes to `\u`-escape control characters, which leads to notable string-bloat when the control characters are utilized - they tend to represent values that appear with high frequency in the encoding.

Reply:
```
> JSON.stringify(String.fromCharCode(1))
'"\\u0001"'
```

With `ENCODED_NUM_BASE=0`...

- note that most leaf notes will have low, single-digit `.entries` counts, all of which would be represented by control codes and thus subject to `\u`-escaping.  The word lengths will usually be notably less than 32 chars and thus would also subject to the same effects... leading to most encoded entries having length less than 32 chars.

- also, most near-leaf internal nodes will have but a few legal `values` leading to child nodes, once again using control codes for their representation.

JSON.stringify use is much prettier and straightforward with `ENCODED_NUM_BASE=0x0020` or above, as this bypasses the control-code range with one exception: 0x007f (DEL).